### ISSUE TITLE
Usage shortcuts of arguments object breaks old version of WebKit

### ISSUE THREAD

**Issue by aaronjensen** [zloirock/core-js#150]
Ok, this is an odd one and is more likely a (fixed) webkit issue and not a core-js issue, but I wanted to point it out here in case you wanted to do something about it.

This change: https://github.com/zloirock/core-js/commit/5bcabcaf7f5425fc7a61610ae82685ef3aae0fce#diff-f68e573b7ecfc93a7afcce675b0459eaL24

Causes this error: 

```
TypeError: 'undefined' is not an object (evaluating '$$[index++]')
```

On this stack in capybara-webkit:

```
Linux id16784 3.8.0-44-generic #66~precise1-Ubuntu SMP Tue Jul 15 04:01:04 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
Capybara: 2.5.0
capybara-webkit: 1.7.1
Qt: 5.0.2
WebKit: 537.21
QtWebKit: 5.1.1
```

But it works fine on OS X with a newer version of Qt:

```
Capybara: 2.5.0
capybara-webkit: 1.7.1
Qt: 5.5.1
WebKit: 538.1
QtWebKit: 5.5.1
```

**Comment by zloirock** [comment]
Interesting. It means this code doesn't work in this WebKit version:
```js
(function(){
  var $$ = arguments;
  return $$[0];
})(42);
```
Can you confirm it?

**Comment by aaronjensen** [comment]
That doesn't appear to cause it. 

I also tried

```js
(function(){ var $$ = arguments, $$len = $$.length, i = 0; while(i<$$len){ $$[i]; i++; } })(42, 43)
```

**Comment by zloirock** [comment]
I have no ideas what else can cause this problem. Can you add some additional info - line, stack, etc?

**Comment by aaronjensen** [comment]
As I mentioned in chat, I can't get much more info. Here is the line causing it:

```js
    const props = {
      ...this.props,
      href: path + wrappedWindow.location.search,
      onClick(e) {
        e.preventDefault()
        navigationActions.changePath(path)
      },
    }
```

It is in a react component that is rendered several times on a page. It's only the last call to it that has the problem, all the others are fine.

**Comment by aaronjensen** [comment]
It happens after maybe 80 calls to it, so it's very likely a broken optimization

**Comment by aaronjensen** [comment]
Great, thanks. Any chance of a 1.2.7 w/ this applied?

**Comment by zloirock** [comment]
It's not a critical bug (a very specific engine), so I don't think makes sense publish `1.2.7`. Today I'll publish `2.0.0-beta.2`, stable `2.0` will be available when I will fix some other problems.

**Comment by aaronjensen** [comment]
Ok, thanks. I don't think I can use 2.0 yet because too many things are pinned to 1.0 but I forked it, so it's ok. Thanks!

**Comment by zloirock** [comment]
Fixed in [2.0](https://github.com/zloirock/core-js/releases/tag/v2.0.0).
